src/states/node.rs

```diff
@@ -1260,7 +1253,17 @@ impl Node {
         Ok(())
     }
 
-    fn handle_group_split(&mut self, prefix: Prefix<XorName>) -> Result<(), RoutingError> {
+    fn handle_group_split(&mut self,
+                          prefix: Prefix<XorName>,
+                          joining_node: XorName)
+                          -> Result<(), RoutingError> {
+        // Send GroupSplit notifications if we don't know of the new node yet
+        if !self.peer_mgr
+            .routing_table()
```

Comment:
If it doesn't have that already, we should really add a `fn has(&self, name: T) -> bool` to `RoutingTable`.